### PR TITLE
DOC: Improve the docsting of Series.iteritems

### PR DESCRIPTION
**Pull request by HubertKl**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -1446,6 +1446,50 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
     def iteritems(self):
         """
         Lazily iterate over (index, value) tuples.
+
+        This method returns a zip of tuples (index, value). This is useful When
```
Comment (TomAugspurger) [inline pandas/core/series.py]: How about "iterable" instead of "zip"?

File: pandas/core/series.py [hunk 2]
```diff
@@ -1446,6 +1446,50 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
...
+        one want to create new series from the values of an old one. Be aware
```
Comment (TomAugspurger) [inline pandas/core/series.py]: "one want**s**" (but "you want" probably sounds better)

Comment (TomAugspurger) [reply pandas/core/series.py]: I'm not sure about creating a new series using itertuples. I'd rather say something about lazily iterating over the index, value pairs.

File: pandas/core/series.py [hunk 3]
```diff
@@ -1446,6 +1446,50 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
...
+        that this might not the fastest way of creating new series.
+
+        Returns
+        -------
+        zip
```
Comment (TomAugspurger) [inline pandas/core/series.py]: I wonder if people have thoughts here. I guess `zip` is technically a type... But I think Iterable is probably better-known.

File: pandas/core/series.py [hunk 4]
```diff
@@ -1446,6 +1446,50 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
...
+            Iterable tuples (index, value) of the Series.
```
Comment (TomAugspurger) [inline pandas/core/series.py]: Iterable of tuples containing the `(index, value)` pairs from a Series.

**Review by TomAugspurger — COMMENTED** [review]
Looks nice overall.

I'm not entirely sure about the focus on "creating a new series". I think the thing to focus on is other functions that can only take scalar inputs (are not vectorized). You can do many things with the output of the non-vectorized method (including creating a new series).

**Comment by HubertKl** [comment]
Thanks for those comments, I tried to improve the docstring accordingly to your remarks.

File: pandas/core/series.py (hunk 2 above)
Comment (HubertKl) [reply pandas/core/series.py]: I am not sure to perfectly get what you mean. I tried to add an example where vectorization is not possible (as far as I know maybe I am wrong)

File: pandas/core/series.py [hunk 5]
```diff
@@ -1446,6 +1446,30 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
     def iteritems(self):
         """
         Lazily iterate over (index, value) tuples.
+
+        This method returns an iterable tuple (index, value). This is
+        convienient if you want to create a a Lazy iterator.
```
Comment (mroeschke) [inline pandas/core/series.py]: There are 2 `a`s on this line. Lowercase l on `Lazy`

File: pandas/core/series.py [hunk 6]
```diff
@@ -1446,6 +1446,30 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
...
+
+        Returns
+        -------
+        iterable
+            Iterable of tuples containing the (index, value) pairs from a
+            Series.
+
+        See Also
+        --------
+        Series.apply : Invoke function on values of Series.
```
Comment (mroeschke) [inline pandas/core/series.py]: Not sure if `Series.apply` and `Series.map` are too related to this method. I would instead reference `Series.items`

Comment (HubertKl) [reply pandas/core/series.py]: Unless I am mistaken, it seems to me that Series.iteritems and Series.items are the same method. The docstring of items and iteritems will be the same I don't think that this is a good idea.

I will mention it in the description instead.

Comment (WillAyd) [reply pandas/core/series.py]: I agree that apply and map aren't closely enough related to include here - can you remove?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Minor comment otherwise lgtm

**Comment by TomAugspurger** [comment]
@HubertKl can you update, removing Series.apply and Series.map in the See Also?

**Comment by jreback** [comment]
@HubertKl can you update to comments

**Comment by HubertKl** [comment]
Sorry for the delay. I modified the PR accordingly to the requests.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @HubertKl !

**Pull request merged**